core/biogeochem/POPLUC.F90

```diff
@@ -1493,17 +1514,27 @@ SUBROUTINE POP_LUC_CASA_transfer(POPLUC, POP, LUC_EXPT, &
 
        POPLUC%HarvProdLoss(g,:)  = kHarvProd * POPLUC%HarvProd(g,:)
        POPLUC%ClearProdLoss(g,:) = kClearProd * POPLUC%ClearProd(g,:)
-       POPLUC%AgProdLoss(g)      = kAgProd * POPLUC%AgProd(g)
+       POPLUC%AgProdLossPast(g)  = kAgProd * POPLUC%AgProdPast(g)
+       POPLUC%AgProdLossCrop(g)  = kAgProd * POPLUC%AgProdCrop(g) 
+       POPLUC%AgProdLoss(g)      = POPLUC%AgProdPast(g) + POPLUC%AgProdCrop(g)  
 
        if (POPLUC%grass(g).gt.0.0_dp .and. l.eq.j+2) then
+          ! Those quantities are calculated at grid cell level (hence multiplied by frac) 
           POPLUC%FAg(g)        = casaflux%Charvest(l)*patch(l)%frac
-          POPLUC%AgProd(g)     = POPLUC%AgProd(g) + casaflux%Charvest(l)*patch(l)%frac - POPLUC%AgProdLoss(g)
-          casaflux%charvest(l) = 0.0_dp
-          casaflux%nharvest(l) = 0.0_dp
-          casaflux%fharvest(l) = min(POPLUC%past(g)/POPLUC%grass(g),1.0_dp) * HIpast + &
-               min(POPLUC%crop(g)/POPLUC%grass(g),1.0_dp) * HIcrop !  fraction grass AGB to be removed next year
-          !write(*,*)'harvest', g,   patch(l)%frac,  casaflux%fharvest(l), POPLUC%crop(g),  POPLUC%past(g)
-          !casaflux%fharvest(l) = 0; ! test vh!
+          POPLUC%AgProdPast(g) = POPLUC%AgProdPast(g) + casaflux%CharvestPast(l) &
+                                 * patch(l)%frac - POPLUC%AgProdLossPast(g)
+          POPLUC%AgProdCrop(g) = POPLUC%AgProdCrop(g) + casaflux%CharvestCrop(l) & 
+                                 * patch(l)%frac - POPLUC%AgProdLossCrop(g)
+          POPLUC%AgProd(g)     = POPLUC%AgProdPast(g) + POPLUC%AgProdCrop(g)
+          casaflux%charvest(l)     = 0.0_dp
```

Comment:
Note to self - the use of `l` here and restriction on hard-wires the POPLUC grass tile to be the last of the patch/tiles in each grid cell

Reply:
yes, this assumption is made in a couple of places in the code. One would need to be really careful to disentangle that.